### DISCUSSION TITLE
Moving to Bun

### DISCUSSION THREAD

**Discussion by VictorVSa** [thheller/shadow-cljs#1147] [Ideas]
Bun 1.0.0 was released on September 8 2023. ([https://bun.sh](https://bun.sh))

By watching their [release video](https://www.youtube.com/watch?v=BsnCpESUEqM) and their [blog post](https://bun.sh/blog/bun-v1.0), it is clearly apparent the pros of this new piece of software that _completely replaces Node.js_, in a drop-in way. 

There are some incompatibilities, as they say, and I'm not sure if those applies for shadow-cljs, but would it be worth a try? 

I've made some tests myself and, for me, it's in a completely different league. It will be very quickly adopted in my opinion.

Could we start looking into either replacing node/npm entirely to use Bun or support using one or the other by a configuration option?

**Comment by thheller** [comment]
What do you expect shadow-cljs would gain from that? `node/npm` is just a delivery mechanism, it is basically not used after that once delivered. If its truly drop-in then you can just use it today with no changes necessary to shadow-cljs. I even expect build ouputs such as `:node-script` or `:esm` to just work.
